modules/command/src/main/java/org/apache/fluo/command/FluoProgram.java

```diff
@@ -0,0 +1,125 @@
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.fluo.command;
+
+import java.util.Optional;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.ParameterException;
+
+import static org.apache.fluo.command.FluoCommandType.CONFIG;
+import static org.apache.fluo.command.FluoCommandType.EXEC;
+import static org.apache.fluo.command.FluoCommandType.GET_JARS;
+import static org.apache.fluo.command.FluoCommandType.INIT;
+import static org.apache.fluo.command.FluoCommandType.LIST;
+import static org.apache.fluo.command.FluoCommandType.ORACLE;
+import static org.apache.fluo.command.FluoCommandType.REMOVE;
+import static org.apache.fluo.command.FluoCommandType.SCAN;
+import static org.apache.fluo.command.FluoCommandType.STATUS;
+import static org.apache.fluo.command.FluoCommandType.WAIT;
+import static org.apache.fluo.command.FluoCommandType.WORKER;
+
+public class FluoProgram {
+  public static void main(String[] args) {
+    FluoConfig fluoConfig = new FluoConfig();
+    FluoExec fluoExec = new FluoExec();
+    FluoGetJars fluoGetJars = new FluoGetJars();
+    FluoInit fluoInit = new FluoInit();
+    FluoList fluoList = new FluoList();
+    FluoOracle fluoOracle = new FluoOracle();
+    FluoRemove fluoRemove = new FluoRemove();
+    FluoScan fluoScan = new FluoScan();
+    FluoStatus fluoStatus = new FluoStatus();
+    FluoWait fluoWait = new FluoWait();
+    FluoWorker fluoWorker = new FluoWorker();
+    JCommander jcommand = JCommander.newBuilder().addCommand(CONFIG.getCommandName(), fluoConfig)
+        .addCommand(EXEC.getCommandName(), fluoExec)
+        .addCommand(GET_JARS.getCommandName(), fluoGetJars)
+        .addCommand(INIT.getCommandName(), fluoInit).addCommand(LIST.getCommandName(), fluoList)
+        .addCommand(ORACLE.getCommandName(), fluoOracle)
+        .addCommand(REMOVE.getCommandName(), fluoRemove).addCommand(SCAN.getCommandName(), fluoScan)
+        .addCommand(STATUS.getCommandName(), fluoStatus).addCommand(WAIT.getCommandName(), fluoWait)
+        .addCommand(WORKER.getCommandName(), fluoWorker).build();
+
+    try {
+      jcommand.parse(args);
+    } catch (ParameterException e) {
+      System.err.println(e.getMessage());
+      String commandName = Optional.ofNullable(jcommand.getParsedCommand()).orElse("");
+      JCommander parsedCommandOrProgram =
+          Optional.ofNullable(jcommand.findCommandByAlias(commandName)).orElse(jcommand);
+      parsedCommandOrProgram.setProgramName(String.format("fluo %s", commandName));
+      parsedCommandOrProgram.usage();
+      System.exit(1);
+      return;
+    }
+
+    FluoCommandType parsedCommandType = FluoCommandType.valueOf(jcommand.getParsedCommand());
+
+    FluoCommand parsedCommand;
+    switch (parsedCommandType) {
```

Comment:
Looking at the Jcommander API I was wondering if it would be possible to replace this switch statement with `Jcommander.getObjects()`

Reply:
I saw that method as well, but I'm not exactly sure how to take advantage of it. My understanding is that `getObjects()` will return a list of the command objects that were passed in via `addCommand`. But we'll still need to select the correct object depending on which command was parsed.

Reply:
I was wondering about it behavior in the following case.

```java
JCommander command = jcommand.findCommandByAlias(parsedCommandType.getCommandName());
var objs = command.getObjects(); //does this return all objs or a single obj related to the specific sub command
```

Reply:
That's a good idea. So that will return a `List<Object>` of size 1, with the object being the one related to the specific sub command.
We can replace the switch statement with something like this
```
JCommander command = jcommand.findCommandByAlias(parsedCommandType.getCommandName());
List<Objects> objs = command.getObjects();
if(objs.size() ==1 && objs.get(0) instanceof FluoCommand) {
  FluoCommand parsedCommand = (FluoCommand)objs.get(0);
  ...
} else {
  //Error Handling
}
```

I think that's definitely more concise than the switch statement, and we may even be able to get rid of the enum completely. It does feel slightly less clean though with the size and instance of check.

Reply:
It the expectation is that its always a FluoCommand and of size one, then the following code would throw an exception when this is not the case w/o the if stmt checking size and instanceof.

```java
  FluoCommand parsedCommand = (FluoCommand)Iterables.getOnlyElement(command.getObjects());
```